ENGAGEHF/Medications/MedicationsList.swift

```diff
@@ -10,29 +10,31 @@ import SwiftUI
 
 
 struct MedicationsList: View {
-    let medications: [MedicationDetails]
+    private let viewModel: ViewModel
     
     
     var body: some View {
-        if !medications.isEmpty {
+        if viewModel.containsRecommendations {
             List {
-                ForEach(medications.sorted(by: { $0.type > $1.type })) { medication in
-                    ExpandableListCard(
-                        label: {
-                            RecommendationSummary(medication: medication)
-                        },
-                        content: {
-                            MedicationRowContent(medication: medication)
-                        }
-                    )
+                if !viewModel.currentlyTakenMedications.isEmpty {
+                    MedicationSection(header: "Current Medications", medications: viewModel.currentlyTakenMedications)
+                }
+                if !viewModel.notCurrentlyTakenMedications.isEmpty {
+                    MedicationSection(header: "Medications That May Help", medications: viewModel.notCurrentlyTakenMedications)
                 }
             }
                 .expandableCardListStyle()
+                .headerProminence(.increased)
         } else {
             ContentUnavailableView("No medication recommendations", systemImage: "pill.fill")
                 .background(Color(.systemGroupedBackground))
         }
     }
+    
+    
+    init(medications: [MedicationDetails]) {
+        self.viewModel = ViewModel(medications)
```

Comment:
This `ViewModel` is being recreated whenever this initializer is called. Make sure to wrap it in an `ObservedObject` property wrapper or make otherwise sure that you do not recreate it on every rerender.

Reply:
Good catch! I also stored the viewModel as a state variable in the relevant Views.